### PR TITLE
practice6

### PR DESCRIPTION
**Pull request by NastasiaIOSdev**

### CODE REVIEW

File: TaskTwoV1/Constants/Constants.swift [hunk 1]
```diff
@@ -10,4 +10,6 @@ import Foundation
 struct Constants {
     static let breedsURL = URL(string: "https://api.thecatapi.com/v1/breeds")
     static let peopleURL = URL(string: "https://swapi.dev/api/people/")
+    static let topHeadlinesURL = URL(string: "https://newsapi.org/v2/top-headlines?country=us&apiKey=bd85a9f5eb6c44aa95856a39ed4eeeea")
```
Comment (vasilevkin) [inline TaskTwoV1/Constants/Constants.swift:13]: better to move api key into separate let constant and use string concatenation.

File: TaskTwoV1/Protocols&Extensions/NewsExtensions/NewsViewController+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        let item = viewModels[indexPath.row]
+        performSegue(withIdentifier: segueIdentifier, sender: item)
+        print("DID select item\(indexPath)")
+    }
+}
+
+extension NewsViewController: UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: collectionView.frame.width / (self.listType == 2 ? 2 : 1) - 10, height: collectionView.frame.height/4 - 15)
```
Comment (vasilevkin) [inline TaskTwoV1/Protocols&Extensions/NewsExtensions/NewsViewController+Extensions.swift:42]: do not use magic numbers

**Review by vasilevkin — CHANGES_REQUESTED** [review]
Good start for Practice 6 , way to go!

File: TaskTwoV1/Constants/Constants.swift (hunk 1 above)
Comment (NastasiaIOSdev) [reply TaskTwoV1/Constants/Constants.swift]: fixed

File: TaskTwoV1/APIService/Model/NewsModel/NewsDataModel.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        apiClient.getNewsInfo(currentPage: currentPage, pageSize: pageSize, filters: chosenFilters, searchQuery: searchQuery) { (response, error) in
+            DispatchQueue.main.async {
+                self.isFerchInProgress = false
+                guard let response = response else {
+                    self.delegate?.onFetchFailed(with: error?.localizedDescription ?? "An error has occured")
+                    return
+                }
+                self.currentPage += 1
+                self.fetchedArticles.append(contentsOf: response.articles)
+                self.totalArticlesCount = response.totalResults < 100 ? response.totalResults : 100
```
Comment (vasilevkin) [inline TaskTwoV1/APIService/Model/NewsModel/NewsDataModel.swift:80]: better move 100 to some constant

**Review by vasilevkin — APPROVED** [review]

File: TaskTwoV1/APIService/Model/NewsModel/NewsDataModel.swift (hunk 3 above)
Comment (NastasiaIOSdev) [reply TaskTwoV1/APIService/Model/NewsModel/NewsDataModel.swift]: fixed

**Pull request merged**